test/System/String.test.cpp

```diff
@@ -74,8 +84,8 @@ TEST_CASE("[System] sf::String")
             CHECK(string.toAnsiString() == ""s);
             CHECK(string.toWideString() == L""s);
             CHECK(string.toUtf8() == std::basic_string<std::uint8_t>());
-            CHECK(string.toUtf16() == std::basic_string<std::uint16_t>());
-            CHECK(string.toUtf32() == std::basic_string<std::uint32_t>());
+            CHECK(string.toUtf16() == u""s);
+            CHECK(string.toUtf32() == U""s);
```

Comment:
Minor thing: there is a `operator==` between `basic_string<T>` and `const T*`, no?
So it's not needed to use the `s` postfix and allocate for each right-hand-side argument.

(same for other occurrences)

Reply:
That's true. The problem then becomes that we need to add even more `doctest::StringMaker` specializations for all those different types of null-terminal strings. We can do that, it's just tedious.

Reply:
Ah, I see, in that case let's keep the tests simpler.

> In either case, I don't think any allocations are occurring since all these strings are so small.

I wasn't aware that small string optimization was common practice, [but it seems to be](https://stackoverflow.com/q/21694302). Thanks!